### ISSUE TITLE
Calendar selecting disabled year/month  (Accessibility)

### ISSUE THREAD

**Issue by mristov** [primefaces/primeng#14637]
### Describe the bug

We can select disabled year or month via tab key even when is disabled (Accessibility)

### Environment

Angular 17
PrimeNg 17.3.3

### Reproducer

https://stackblitz.com/edit/sz4dye?file=src%2Fapp%2Fdemo%2Fcalendar-reactive-forms-demo.ts

### Angular version

17.0.5

### PrimeNG version

17.3.3

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

18.16.1

### Steps to reproduce the behavior

In Stackblitz link we can clearly see that if we set **maxDate** 
-> open _p-datepicker-header_
-> via click we can't click on **disabled** years, but if we press Tab key we can focus in disabled years and then with Space key or Enter key we can select it.
-> with selected disabled year we are opening months for that year, and again as part of **Accessibility** if we press Tab key focus is losing outside datepicker

This clearly cannot happen

The same scenario happens also and for selecting disabled months.

<img width="213" alt="image" src="https://github.com/primefaces/primeng/assets/150306240/2284edef-edcf-41b5-95c9-d2ad84f16142">
<img width="194" alt="image" src="https://github.com/primefaces/primeng/assets/150306240/2068ddd6-536a-48f4-beb6-34aa1748ed2f">

### Expected behavior

As we cannot click on disabled year/month, the same applies and for press Enter key or Space key over disabled.
